Sources/Data Model/Experiment.swift

```diff
@@ -63,4 +75,63 @@ extension Experiment {
     var isActivated: Bool {
         return status == .running
     }
+    
+    mutating func serializeAudiences(with audiencesMap: [String: String]) {
+        guard let conditions = audienceConditions else { return }
+        
+        let serialized = conditions.serialized
+        audiences = replaceAudienceIdsWithNames(string: serialized, audiencesMap: audiencesMap)
+    }
+    
+    /// Replace audience ids with audience names
+    ///
+    /// example:
+    /// - string: "(AUDIENCE(1) OR AUDIENCE(2)) AND AUDIENCE(3)"
+    /// - replaced: "(\"us\" OR \"female\") AND \"adult\""
+    ///
+    /// - Parameter string: before replacement
+    /// - Returns: string after replacement
+    func replaceAudienceIdsWithNames(string: String, audiencesMap: [String: String]) -> String {
```

Comment:
can we have unit test for this method.

Reply:
We have a unit test case covering this audience mapping - https://github.com/optimizely/swift-sdk/blob/d91c49c41a9b921604527a4beb6aa9c1c08f105d/Tests/OptimizelyTests-DataModel/ExperimentTests.swift#L152